aes_test.go

```diff
@@ -1,27 +1,35 @@
 //go:build linux
 
-package openssl
+package openssl_test
 
 import (
 	"bytes"
 	"crypto/cipher"
 	"math"
 	"testing"
+
+	"github.com/golang-fips/openssl/v2"
 )
 
 func TestNewGCMNonce(t *testing.T) {
 	key := []byte("D249BF6DEC97B1EBD69BC4D6B3A3C49D")
-	ci, err := NewAESCipher(key)
+	ci, err := openssl.NewAESCipher(key)
 	if err != nil {
 		t.Fatal(err)
 	}
-	c := ci.(*aesCipher)
 
-	gi, err := c.NewGCM(gcmStandardNonceSize, gcmTagSize)
+	const (
+		gcmTagSize           = 16
+		gcmStandardNonceSize = 12
+	)
+
+	c := ci.(interface {
+		NewGCM(nonceSize, tagSize int) (cipher.AEAD, error)
+	})
+	g, err := c.NewGCM(gcmStandardNonceSize, gcmTagSize)
```

Comment:
The type is an implementation details that I would prefer not to export. The idea is to test the `openssl` package as an external consumer, which won't have access to `aesCipher` but would have to type-cast the interface.

About the constants, it would be helpful to export them in `export_test.go`, but the idiom is to just export what's indispensable for testing (ideally nothing). If something can be recreated from outside the tested package, as an external consumer would, then better not export it.